src/common_node.cpp

```diff
@@ -19,3 +20,85 @@ void CommonNode::heartbeat_timer_callback() {
     heartbeat_publisher->publish(message);
     RCLCPP_DEBUG(this->get_logger(), "CommonNode::heartbeat_timer_callback: Published heartbeat message with sender_id: %s, tick: %u, active: %d", this->get_fully_qualified_name(), message.tick, message.active);
 }
+
+/**
+ * @brief Handles the completion of a job.
+ *
+ * This function sends a job_finished message with the given error code and payload.
+ * Additionally, deactivates the node.
+ *
+ * @param error_code The error code associated with the job completion (EXIT_SUCCESS == 0 if no error).
+ * @param payload The payload data associated with the job completion.
+ */
+void CommonNode::job_finished(const uint8_t error_code, const nlohmann::json &payload)
+{
+    interfaces::msg::JobFinished msg;
+
+    msg.sender_id = this->get_fully_qualified_name();
+    msg.error_code = error_code;
+    msg.payload = payload.dump();
```

Comment:
Hier könnte auch eine exception geworfen werden.
```suggestion
    try
    {
        msg.payload = payload.dump();
    }
    catch (const nlohmann::json::parse_error &e)
    {
        RCLCPP_FATAL(this->get_logger(), "CommonNode::job_finished: Payload is not a valid JSON. Stopping node.");
        exit(EXIT_FAILURE);
    }
```

Reply:
Könnte man grundsätzlich machen, aber ich denke, wenn hier irgendwas schief gelaufen ist, ist das sowieso ein Programmierfehler und obs jetzt in der Funktion explodiert oder beim Aufrufer macht eigentlich keinen Unterschied.